roles/wordpress-setup/templates/ssl.no-default.conf.j2

```diff
@@ -0,0 +1,38 @@
+# {{ ansible_managed }}
+
+# Drop requests for unknown hosts
+#
+# If no default server is defined, nginx will use the first found server.
+# To prevent host header attacks, or other potential problems when an unknown
+# servername is used in a request, it's recommended to drop the request
+# returning 444 "no response".
+
+{% set _item = {
+    'key': ssl_default_cert_site | default(sites_using_ssl[0]),
+    'value': wordpress_sites[ssl_default_cert_site | default(sites_using_ssl[0])]
+  }
+-%}
+
+server {
+  listen [::]:443 ssl default_server deferred;
+  listen 443 ssl default_server deferred;
+
+  include h5bp/directive-only/ssl.conf;
+  ssl_dhparam /etc/nginx/ssl/dhparams.pem;
+
+  {% if _item.value.ssl.provider | default('manual') == 'manual' and _item.value.ssl.cert is defined and _item.value.ssl.key is defined -%}
+  ssl_certificate         {{ nginx_path }}/ssl/{{ _item.value.ssl.cert | basename }};
+  ssl_certificate_key     {{ nginx_path }}/ssl/{{ _item.value.ssl.key | basename }};
+
+  {% elif _item.value.ssl.provider | default('manual') == 'letsencrypt' -%}
+  ssl_certificate         {{ nginx_path }}/ssl/letsencrypt/{{ _item.key }}-{{ letsencrypt_cert_ids[_item.key] }}-bundled.cert;
+  ssl_certificate_key     {{ nginx_path }}/ssl/letsencrypt/{{ _item.key }}.key;
+
+  {% elif _item.value.ssl.provider | default('manual') == 'self-signed' -%}
+  ssl_certificate         {{ nginx_path }}/ssl/{{ _item.key }}.cert;
+  ssl_trusted_certificate {{ nginx_path }}/ssl/{{ _item.key }}.cert;
+  ssl_certificate_key     {{ nginx_path }}/ssl/{{ _item.key }}.key;
+  {% endif -%}
+
```

Comment:
Does it return `444` if we `include includes.d/{{ item.key }}/*.conf;` here? The goal is to allow third party roles to inject `ssl_certificate` and `ssl_certificate_key` here.
```
  {% block includes_d -%}
  include includes.d/{{ item.key }}/*.conf;
  {% endblock -%}
```

Reply:
>By default, it will pick the cert and key from one of the ssl-enabled sites. However, users may specify which site's cert and key to use in group_vars:
`ssl_default_cert_site: example.com`

So, users can specify which site's cert and key to use from `wordpress_sites`. But it seems that you anticipate that users or third-party roles may want to specify a cert and key other than any in `wordpress_sites`. I guess it's possible. I hadn't considered it.

My first idea to accommodate this scenario would be add to this PR the vars `ssl_certificate_default` and `ssl_certificate_key_default` for users or roles to define. In that case, this `ssl.no-default.conf` could use those values if defined or fall back on the cert and key for a site from `wordpress_sites`, as in the code above.

Considerations for integration with third-party roles are important but are not fresh in my mind. Would vars `ssl_certificate_default` and `ssl_certificate_key_default` be sufficient? Or does the `include includes.d/...` approach seem superior? 

My first inclination is to avoid adding `include includes.d/...` to this `ssl.no-default.conf`. Given that this server block is filling requests for unknown hosts, my sense is that it should just get rid of the request with as little processing as possible (i.e., avoid all the potential processing of confs in `includes.d`). If there are examples of strong use cases for includes here, perhaps we should give users the option to specify includes paths:

```
# some group_vars location
# corresponding templates in local `nginx-includes` directory
nginx_no_default_include_paths:
  - includes.d/no-default/foo.conf
```

```
# in `no-default.conf.j2` and `ssl.no-default.conf.j2`
{% for path in nginx_no_default_include_paths | default([]) %}
include {{ path }}
{% endfor %}
```

I'm reluctant to add all this complexity unless it is likely to be used. But if likely, then :+1:

Reply:
I am over-concerned because this block only runs for *unknown* domains. Thus, we can't give it a valid ssl certificate for most of the cases. 

Thanks!

Reply:
Questions:
- why not `ssl_verify_client off; ` explicitly meaning?
- why skipping `ssl_certificate`, `ssl_trusted_certificate` and `ssl_certificate_key` and let Nginx get them form  another server block?

Reply:
Given the Nginx [default `ssl_verify_client off;`](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_verify_client), does it help to add the explicit directive?

When those other directives were absent, the real ssl-enabled `wordpress_sites` seemed to fail:
> Without `ssl_certificate` and `ssl_certificate_key` in the default `server` block, https sites fail:
>  * in browser: "The connection was reset" `ERR_CONNECTION_RESET`
>  * curl: `curl: (35) Server aborted the SSL handshake`

In any case, I don't think those other server blocks would match an unknown host.

If there were only one site, never a need for different certs/keys for different hosts, then we could specify the cert and key just once in an `http` block. But I assume we want to enable multiple ssl-enabled `wordpress_sites` with separate certs.

These issues are new to me however, so I'm eager for suggestions.

Reply:
About `ssl_verify_client off;`, over-concerned again, sorry.

> When those other directives were absent, the real ssl-enabled `wordpress_sites` seemed to fail

Those other directives are absent even with this patch. Edge case: All sites using third party ssl providers.

Possible solution: Always generate self-signed cert for `ssl.no-default.conf` as https://github.com/h5bp/server-configs-nginx/pull/177 suggested.

This opens another question: Is it good to enable `ssl.no-default.conf` even no site using SSL?

Reply:
> Is it good to enable `ssl.no-default.conf` even no site using SSL?

I imagine the answer is "no," which is why this PR adds `ssl.no-default.conf.j2` to the [`nginx_sites_confs`](https://roots.io/trellis/docs/nginx-includes/#sites-templates) list with the option `enabled: "{{ sites_using_ssl != [] }}"`. The result is that `ssl.no-default.conf` is only symlinked into `sites-enabled` when at least one of the `wordpress_sites` has SSL enabled.

I hadn't considered the edge case that someone would get a third party cert and key on the server through means other than the Trellis built-in Let's Encrypt, [manual](https://roots.io/trellis/docs/ssl/#manual), or self-signed options. If it were common, perhaps always adding a self-signed cert to `ssl.no-default.conf` would be good, or it could be an argument for adding the `ssl_certificate_default` and `ssl_certificate_key_default` vars mentioned above.

I figure that such would be a rare edge case, however, and this PR in its current form still allows users to customize the list of `nginx_sites_confs`, optionally replacing the default `ssl.no-default.conf.j2` with `some-custom-no-default.conf.j2`.